File: backend/routes/department.go
package routes

import (
	"oa-system/handlers"

	"github.com/gin-gonic/gin"
)

// departmentRoutes 执行相关业务逻辑
func departmentRoutes(rg *gin.RouterGroup) {
	rg.GET("/departments", handlers.GetDepartments)
	rg.POST("/departments", handlers.CreateDepartment)
	rg.PUT("/departments/:id", handlers.UpdateDepartment)
	rg.DELETE("/departments/:id", handlers.DeleteDepartment)
	rg.GET("/departments/:id/menu-permissions", handlers.GetDepartmentMenuPermissions)
	rg.PUT("/departments/:id/menu-permissions", handlers.SetDepartmentMenuPermissions)
}

File: frontend/src/api/department.js
import http from './http'

export const getDepartments = (params) => http.get('/departments', { params })
export const createDepartment = (data) => http.post('/departments', data)
export const updateDepartment = (id, data) => http.put(`/departments/${id}`, data)
export const deleteDepartment = (id) => http.delete(`/departments/${id}`)
export const getDepartmentMenuPermissions = (id) => http.get(`/departments/${id}/menu-permissions`)
export const setDepartmentMenuPermissions = (id, data) => http.put(`/departments/${id}/menu-permissions`, data)
